frontend/src/metabase/query_builder/components/filters/FilterWidget.jsx

```diff
@@ -135,6 +137,13 @@ export default class FilterWidget extends Component {
                     isInitiallyOpen={this.props.filter[1] === null}
                     onClose={this.close}
                     horizontalAttachments={["left"]}
+                    autoWidth={
+                        // If the field is a date field the filter popover
+                        // should be allowed to auto size its width to accomodate
+                        // the calendars
+                        // $FlowFixMe
+                        dimension.field().isDate()
```

Comment:
@attekei Running into one issue here. This only evaluates to true if the filter has actually been added, so if you're adding a new filter the issue still persists. Any ideas on how to properly handle that?

Reply:
Ah, I think that I can take a look tomorrow and push a commit that fixes that.

I like the `autoWidth` prop! I could merge this PR after pushing the fix if that is fine to you.

Reply:
Sounds great. Thanks!